pennylane/template.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+r"""
+QML Templates
+=============
+
+**Module name:** :mod:`pennylane.template`
+
+.. currentmodule:: pennylane.template
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build
+more complex quantum machine learning models.
+
+For example, you can construct, evaluate, and train quantum circuit based
+on the circuit-centric quantum classifier architecture from
```

Comment:
To me, "circuit-centric quantum classifier" is a good name for a paper, but an awkward name for a template/ansatz. It doesn't reveal any specifics about the model except that it is a classifier implemented on a quantum circuit. Can we think of any alternative names?

Reading the "Circuit-centric quantum classifiers" paper more closely, the specific circuit ansatz is called "strongly entangling circuits". Maybe something like "StronglyEntanglingClassifier"?

Reply:
The I would go for `StronglyEntanglingCircuit()` as the template only contains the circuit, not the whole classifier.

Reply:
Sounds good.